dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java

```diff
@@ -245,4 +252,87 @@ private static HTMLPageAsset createPage(){
+
+        final List<ContainerStructure> csList = new ArrayList<ContainerStructure>();
+        final ContainerStructure cs = new ContainerStructure();
+        cs.setStructureId(contentGenericType.id());
+        cs.setCode("$!{body}");
+        csList.add(cs);
+
+        container = APILocator.getContainerAPI().save(container, csList, host, systemUser, false);
+        PublishFactory.publishAsset(container, systemUser, false, false);
+
+        Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
+        Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", false);
+
+        final Contentlet page = new HTMLPageDataGen(host, template).host(host).languageId(1).nextPersisted();
+
+        final Contentlet contentlet = new ContentletDataGen(contentGenericType.id())
+                .languageId(1)
+                .host(host)
+                .setProperty("title", "content1")
+                .setProperty("body", "content1")
+                .nextPersisted();
+
+        ContentletDataGen.publish(contentlet);
+        ContentletDataGen.publish(page);
+
+        new MultiTreeDataGen()
+                .setContainer(container)
+                .setPage((HTMLPageAsset) page)
+                .setContentlet(contentlet)
+                .nextPersisted();
+
+        final HttpServletRequest mockRequest = new MockSessionRequest(
+                new MockAttributeRequest(new MockHttpRequest(host.getName(), ((HTMLPageAsset) page).getURI()).request()).request()
+        )
+                .request();
+
+
+        Mockito.when(mockRequest.getParameter("host_id")).thenReturn(host.getIdentifier());
+        mockRequest.setAttribute(WebKeys.HTMLPAGE_LANGUAGE, "1");
+        HttpServletRequestThreadLocal.INSTANCE.setRequest(mockRequest);
+        mockRequest.getSession().setAttribute("tm_host" , host);
+        mockRequest.getSession().setAttribute(WebKeys.HTMLPAGE_LANGUAGE, "1");
+
+        final HttpServletResponse mockResponse = mock(HttpServletResponse.class);
+
+        final ServletOutputStream outputStream = mock(ServletOutputStream.class);
+        when(mockResponse.getOutputStream()).thenReturn(outputStream);
+
+        Calendar tomorrow = Calendar.getInstance();
+        tomorrow.add(Calendar.DATE, 1);
+
+        //setting Time Machine
+        mockRequest.getSession().setAttribute("tm_date", Long.toString(tomorrow.getTime().getTime()));
+        mockRequest.getSession().setAttribute("tm_lang", "1");
+
+        mockRequest.setAttribute(com.liferay.portal.util.WebKeys.USER, systemUser);
+        mockRequest.getSession().setAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION, PageMode.EDIT_MODE);
+
+        FilterChain chain = Mockito.mock(FilterChain.class);
+        final TimeMachineFilter timeMachineFilter = new TimeMachineFilter();
+        timeMachineFilter.doFilter(mockRequest, mockResponse, chain);
+
+        VelocityServlet velocityServlet = new VelocityServlet();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'velocityServlet' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5474742)